intake/source/base.py

```diff
@@ -43,6 +43,14 @@ class DataSource(object):
     datashape = None
     description = None
 
+    @property
+    def cache_dirs(self):
+        return [c._cache_dir for c in self.cache]
+    
+    def set_cache_dir(self, cache_dir):
```

Comment:
Should be a real setter?

Reply:
I couldn't get a real setter to work.  I was wondering if some of the other code in this class was causing an issue with it.